### ISSUE TITLE
Support resource request tuning for cost optimistation

### ISSUE THREAD

**Issue by alexec** [argoproj/argo-cd#6989]
# Summary

Enhance the node view with information highlighting pods that are using much more or less that they request.

# Motivation

I want to cost optimise and quickly. Nothing I've yet used makes that possible. But Argo CD is well place to make this easy.

# Proposal

![image](https://user-images.githubusercontent.com/1142830/129385377-0b2086e1-b524-4f77-8e91-65010f05aff7.png)

1. This view only appears to surface total requests. What about per-pod requests.
2. Combine this with metrics server data to surface the actual usage.
3. Highlight diffs.

**Comment by keithchong** [comment]
Hi @Alexc, @rbreeze , @alexmt, could you please elaborate on points 2 and 3? 

So for point 1, you want to show the actual CPU/Mem usage in the tooltip of each pod, something like this?

<img width="427" alt="Screen Shot 2021-09-27 at 9 17 36 AM" src="https://user-images.githubusercontent.com/7726022/134919154-4c1dbc70-9fce-4940-b667-02faaf9a0002.png">

Looks like there might be server and app controller changes

**Comment by alexec** [comment]
If a pod is using less resources that it requests (in resource requests) then it is costing more than it needs to. 

Metrics Server stores the resource usage of pods. 

https://github.com/kubernetes-sigs/metrics-server#use-cases

You can then calculate "(requested resources - resource usage) / requested resources" which would tell you how much more a resource is costing to run than it needs to.

You can then adjust your resource requests to pack you pods more efficiently and save $£¥.

**Comment by keithchong** [comment]
Yeah, it looks like in the app controller, we are totalling up the capacity and resource requests for the entire node.  I think we need to keep track of the individual pod stats, right @rbreeze ?

**Comment by keithchong** [comment]
<img width="381" alt="Usage" src="https://user-images.githubusercontent.com/7726022/139683502-9136d7af-cd92-45b8-94a9-69dacd080466.png">

**Comment by alexec** [comment]
Maybe take a look at [Kubecost](https://www.kubecost.com). 

They surface these details, but as a user, I want to know where I can save the most money. I don't want to tweak a resource from 110m to 100m, but I would if it was from 1000m to 500m.

**Comment by keithchong** [comment]
Hi Alex, perhaps integrating this with Kubecost could be a follow-on change? Here we have the individual pod stats. 

<img width="423" alt="Screen Shot 2021-11-05 at 8 57 37 PM" src="https://user-images.githubusercontent.com/7726022/140773657-9bbe2e6e-de6f-47a6-8648-447eff9ff8e1.png">

<img width="471" alt="Screen Shot 2021-11-05 at 8 57 30 PM" src="https://user-images.githubusercontent.com/7726022/140774058-44332534-da3f-40cd-b009-71419615f24c.png">

**Comment by keithchong** [comment]
Latest screenshots with PR:

<img width="365" alt="sync-waves-backend-pod" src="https://user-images.githubusercontent.com/7726022/142508474-dcc6dadb-5427-438b-9b0f-7d2d5c12ca75.png">

<img width="362" alt="sync-waves-frontend-pod" src="https://user-images.githubusercontent.com/7726022/142508477-3290d2ab-4fab-4cf4-b0ea-dd0dd5cf9f48.png">

<img width="346" alt="sync-waves-memory-requests" src="https://user-images.githubusercontent.com/7726022/142508478-339c60e7-c851-4c59-b63e-fa360947d560.png">

<img width="358" alt="sync-waves-cpu-requests" src="https://user-images.githubusercontent.com/7726022/142508475-78f96dc5-0443-4dfb-b38b-449f8fc6c391.png">
